[EDU-4913] [EDU-4912] feature: add query connected users data guide (#1180)
aziontech/docs: src/content/docs/en/pages/guides/graphql/connected-users-dataset.mdx

---
title: How to query Connected Users data with GraphQL API

> description: This guide will explain how to query Connected Used data using the GraphiQL playground.

Author: ```suggestion
description: This guide will explain how to query Connected Users data using the GraphiQL playground.
```

meta_tags: graphql, api, query, postman, live ingest, live streaming
namespace: docs_guides_query_connected_users_graphql
permalink: /documentation/products/guides/query-connected-users-data-with-graphql/
menu_namespace: graphqlMenu
---

The **connectedUsers** dataset lets you get real-time aggregated data, provided by Azion Live Ingest, related to the number of users currently connected to your applications with live streamings. This dataset is part of the Real-Time Metrics GraphQL API and is generated from the raw HTTP events, which are populated just for clients with live streaming configurations.

This information can be accessed through the GraphQL API, allowing you to transfer this data to a third-party platform, and enabling you to further analysis and review of user activity and engagement. Additionally, the data is available for up to *2 years*.

> This guide will explain how to query Connected Used data using the GraphiQL playground.

Reviewer: ```suggestion
This guide will explain how to query Connected Users data using the GraphiQL playground.
```

---

## Querying data by host

To query your data by host, proceed as follows:

1. Access the [GraphiQL playground](https://manager.azion.com/metrics/graphql)
    - You must be logged in to your Azion account. Otherwise, you'll receive an error message.
2. Send a query following this format:

```bash
query ConnectedUsers {
    connectedUsersMetrics(
        limit: 10000,
        filter: {
            tsRange:{
                begin: "2024-04-10T00:00:00",
                end: "2024-04-11T00:00:00"
            }
        },
      	 groupBy: [ts,host]
        orderBy: [ts_DESC]
    ) {
        ts,
	 host,
        uniqueSessions
    }
}
```

Where:

| Field | Description |
|----------|----------|
| `limit` | Specifies the maximum number of results to return. System maximum: `10000` |
| `filter` | Defines the criteria used to filter the data returned by the query |
| `tsRange` | A subfield of `filter`. Specifies a time range for filtering data. It includes `begin` and `end` fields to define the start and end timestamps. Format: `"YYYY-MM-DDTHH:mm:ss"`; example: `"2024-04-11T00:00:00"` |
| `groupBy` | Specifies the fields by which the query results should be grouped. Example: `[ts]` |

> | `orderBy` | Specifies the order in which the results should be returned. Accepted values: `[ts_DESC]`, for descending order, and `[ts_ASC]`, for ascending order |

Reviewer: suggestion: Talvez algo assim pois os valores aceitos podem ser vários.
```suggestion
| `orderBy` | Specifies the order in which the results should be returned. Examples: `[ts_DESC]`, for descending order, and `[ts_ASC]`, for ascending order |
```
